ydb/public/sdk/cpp/examples/vector_index/vector_index.cpp

```diff
@@ -0,0 +1,294 @@
+        last = curr;
+    };
+    auto waitRequest = [&](auto& request) {
+        auto r = request.ExtractValueSync();
+        if (!r.IsSuccess()) {
+            ythrow TVectorException{r};
+        }
+        ++current;
+        if (auto curr = std::chrono::steady_clock::now(); (curr - last) >= std::chrono::seconds{1}) {
+            report(curr);
+        }
+    };
+
+    std::deque<TAsyncStatus> requests;
+    auto waitFirst = [&] {
+        if (requests.size() < std::thread::hardware_concurrency()) {
+            return;
+        }
+        waitRequest(requests.front());
+        requests.pop_front();
+    };
+
+    TParamsBuilder paramsBuilder;
+    TRetryOperationSettings retrySettings;
+    retrySettings
+        .MaxRetries(60)
+        .GetSessionClientTimeout(TDuration::Seconds(60))
+        .Idempotent(true)
+        .RetryUndefined(true);
+    for (ui64 i = 0; i < options.Rows; i += kBulkSize) {
+        waitFirst();
+        paramsBuilder.AddParam("$begin").Uint64(i).Build();
+        paramsBuilder.AddParam("$rows").Uint64(kBulkSize).Build();
+        auto f = client.RetryOperation([&, p = paramsBuilder.Build()](TSession session) {
+            auto params = p;
+            return session.ExecuteDataQuery(
+                              query,
+                              TTxControl::BeginTx(TTxSettings::SerializableRW()).CommitTx(),
+                              std::move(params))
+                .Apply([](auto result) -> TStatus {
+                    return result.ExtractValueSync();
+                });
+        }, retrySettings);
+        requests.push_back(std::move(f));
+    }
+
+    for (auto& request : requests) {
+        waitRequest(request);
+    }
+    report(std::chrono::steady_clock::now());
+}
+
+void TopKFlat(TTableClient& client, const TOptions& options, std::string_view type) {
+    TString query = std::format(R"(
+          $TargetBinary = Knn::ToBinaryStringFloat($Target);
+          $TargetSQ = Knn::ToBinaryString{7}(CAST($Target AS List<{8}>));
+        
+          $IndexIds = SELECT {1}, Knn::{0}({4}, $TargetSQ) as distance
+            FROM {2}
+            ORDER BY distance
+            LIMIT {6} * 2;
+        
+          SELECT Knn::{0}({4}, $TargetBinary) as distance, {5}
+            FROM {3}
+            WHERE {1} IN (SELECT {1} FROM $IndexIds)
+            ORDER BY distance
+            LIMIT {6};
+        )",
+                                options.Distance,
+                                options.PrimaryKey,
+                                IndexName(options),
+                                options.Table,
+                                options.Embedding,
+                                options.Data,
+                                options.TopK,
+                                type,
+                                type == NQuantizer::Bit ? "Float" : type);
+    Cout << query << Endl;
```

Comment:
It's debug output?

Reply:
It depends, in general yes.
But from other side, I think it's nice to user of examples to see what query will be send